packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -276,7 +276,13 @@ class CupertinoDatePicker extends StatefulWidget {
     this.use24hFormat = false,
     this.dateOrder,
     this.backgroundColor,
-  }) : initialDateTime = initialDateTime ?? DateTime.now(),
+  }) : initialDateTime = initialDateTime ?? DateTime(
+      DateTime.now().year,
+      DateTime.now().month,
+      DateTime.now().day,
+      DateTime.now().hour,
+      DateTime.now().minute % minuteInterval * minuteInterval
+    ),
```

Comment:
Maybe there can be one call to DateTime.now() here for perf reasons.

Reply:
Also maybe worth to consider constraining the minute interval, possibly a separate change though.